src/Persistence/MassTransit.Azure.Cosmos.Table.Tests/Audit/AuditStore_Specs.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace MassTransit.Azure.Cosmos.Table.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using GreenPipes.Util;
+    using NUnit.Framework;
+    using Shouldly;
+
+
+    [TestFixture]
+    public class Saving_audit_records_to_the_audit_store :
+        AzureCosmosTableInMemoryTestFixture
+    {
+        IEnumerable<AuditRecord> _records;
+
+        [Test]
+        public async Task Should_Have_Audit_Records()
+        {
+            _records = AzureTableHelpers.GetAuditRecords();
+            _records.ShouldNotBeEmpty();
+        }
+
+        [Test]
+        public async Task Should_have_consume_audit_records()
+        {
+            var consumeRecords = _records.Where(x => x.ContextType == "Consume");
+            consumeRecords.Count().ShouldBe(2);
+        }
+
+        [Test]
+        public async Task Should_have_send_audit_record()
+        {
+            var consumeRecords = _records.Where(x => x.ContextType == "Send");
+            consumeRecords.Count().ShouldBe(2);
+        }
+
+        Task<ConsumeContext<A>> _handledA;
+        Task<ConsumeContext<B>> _handledB;
+
+        [OneTimeSetUp]
+        public async Task SetUp()
+        {
+            await InputQueueSendEndpoint.Send(new A());
+            await _handledA;
+            await InputQueueSendEndpoint.Send(new B());
+            await _handledB;
+        }
+
+        protected override void ConfigureInMemoryBus(IInMemoryBusFactoryConfigurator configurator)
+        {
+            configurator.UseAzureCosmosTableAuditStore(ConnectionString, AuditTableName);
+            base.ConfigureInMemoryBus(configurator);
+        }
+
+        protected override void ConfigureInMemoryReceiveEndpoint(IInMemoryReceiveEndpointConfigurator configurator)
+        {
+            configurator.Consumer<TestConsumer>();
+            _handledA = Handled<A>(configurator);
+            _handledB = Handled<B>(configurator);
+            EndpointConvention.Map<A>(new Uri($"{configurator.InputAddress}"));
```

Comment:
avoid using `EndpointConvention` in tests, you have either `InputQueueAddress` or  `InputQueueSendEndpoint`